File: AuthService/src/main/java/br/com/server/auth/secutiry/UserDetailsServiceImpl.java
package br.com.server.auth.secutiry;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.AuthorityUtils;
import org.springframework.security.core.userdetails.User;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.stereotype.Service;

import java.util.Arrays;
import java.util.List;

@Service
public class UserDetailsServiceImpl implements UserDetailsService {

    public static final String USER = "USER";
    public static final String ADMIN = "ADMIN";

    @Autowired
    private BCryptPasswordEncoder encoder;

    @Override
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {

        final List<AppUser> users = provicesListOfAppUsers();

        for (AppUser user : users) {
            if (user.getUsername().equals(username)) {

                List<GrantedAuthority> grantedAuthorities = AuthorityUtils
                        .commaSeparatedStringToAuthorityList("ROLE_" + user.getRole());

                return new User(user.getUsername(), user.getPassword(), grantedAuthorities);
            }
        }
        throw new UsernameNotFoundException("Username: " + username + " not found");
    }

    private List<AppUser> provicesListOfAppUsers() {
        return Arrays.asList(
                new AppUser(1, "matheus", encoder.encode("12345"), USER),
                new AppUser(1, "admin", encoder.encode("12345"), ADMIN)
        );
    }

    private static class AppUser {
        private Integer id;
        private String username, password;
        private String role;

        public AppUser(Integer id, String username, String password, String role) {
            this.id = id;
            this.username = username;
            this.password = password;
            this.role = role;
        }

        public Integer getId() {
            return id;
        }

        public void setId(Integer id) {
            this.id = id;
        }

        public String getUsername() {
            return username;
        }

        public void setUsername(String username) {
            this.username = username;
        }

        public String getPassword() {
            return password;
        }

        public void setPassword(String password) {
            this.password = password;
        }

        public String getRole() {
            return role;
        }

        public void setRole(String role) {
            this.role = role;
        }
    }
}

File: AuthService/src/main/kotlin/br/com/server/auth/security/UserDetailsServiceImpl.kt
package br.com.server.auth.security

import br.com.server.auth.UserRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.core.authority.AuthorityUtils
import org.springframework.security.core.userdetails.User
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service

@Service
class UserDetailsServiceImpl : UserDetailsService {

    @Autowired
    private lateinit var userRepository: UserRepository

    override fun loadUserByUsername(username: String?): UserDetails {

        for (user in userRepository.getUsers()) {
            if (user.username == username) {
                val grantedAuthorities = AuthorityUtils.commaSeparatedStringToAuthorityList("ROLE_${user.role}")
                return User(user.username, user.password, grantedAuthorities)
            }
        }
        throw UsernameNotFoundException("Username: $username not found")
    }

}